### PR TITLE
schema/config_test.go: add a test for an env var without `=value`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Related: https://github.com/moby/moby/pull/33557

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

### CODE REVIEW

File: schema/validator.go [hunk 1]
```diff
@@ -199,6 +199,13 @@ func validateConfig(r io.Reader) error {
 
 	checkPlatform(header.OS, header.Architecture)
 
+	envRegexp := regexp.MustCompile(`^.+=.*$`)
```
Comment (stevvooe) [inline schema/validator.go]: Should be: ``^[^=]+=.*$`.

**Comment by AkihiroSuda** [comment]
updated @stevvooe

**Comment by AkihiroSuda** [comment]
CI failure is unrelated https://github.com/opencontainers/image-spec/issues/697

**Comment by stevvooe** [comment]
LGTM

**Comment by stevvooe** [comment]
@AkihiroSuda Could you try rebasing?

**Comment by AkihiroSuda** [comment]
rebased

**Comment by philips** [comment]
LGTM

**Comment by vbatts** [comment]
LGTM

**Pull request merged**